label_namer.go

```diff
@@ -49,20 +50,41 @@ type LabelNamer struct {
 //	namer.Build("http.method")     // "http_method"
 //	namer.Build("123invalid")      // "key_123invalid"
 //	namer.Build("__reserved__")    // "__reserved__" (preserved)
-func (ln *LabelNamer) Build(label string) string {
+func (ln *LabelNamer) Build(label string) (normalizedName string, err error) {
+	defer func() {
+		if len(normalizedName) == 0 {
+			err = errors.New("label name normalization resulted in empty name")
+			return
+		}
+
+		if ln.UTF8Allowed || normalizedName == label {
+			return
+		}
+
+		// Check that the resulting normalized name contains at least one non-underscore character
+		for _, c := range normalizedName {
+			if c != '_' {
+				return
+			}
+		}
+		normalizedName = ""
+		err = errors.New("label name normalization resulted in invalid name")
```

Comment:
It would be nice to add the before and after to this message.
```suggestion
		err = fmt.Errorf("normalization for label name %q resulted in invalid name %q", label, normalizedName)
		normalizedName = ""
```